verifiers/internal/gha/slsaprovenance/v0.2/provenance.go

```diff
@@ -0,0 +1,98 @@
+package v02
+
+import (
+	"fmt"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	serrors "github.com/slsa-framework/slsa-verifier/v2/errors"
+	"github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance"
+)
+
+// TODO(asraa): Use a static mapping.
+//
+//nolint:gochecknoinits
+func init() {
+	slsaprovenance.ProvenanceMap.Store(
+		"https://slsa.dev/provenance/v0.2",
+		New)
+}
+
+type ProvenanceV02 struct {
+	*intoto.ProvenanceStatement
+}
+
+func New() slsaprovenance.Provenance {
+	return &ProvenanceV02{}
+}
+
+func (prov *ProvenanceV02) BuilderID() string {
+	return prov.Predicate.Builder.ID
```

Comment:
can the value be nil and panic? Should every function return a `(type, error)`?
Same question for other functions.

Reply:
This one I can add a panic condition, although we do require a builder.id no matter what predicate type. So this one is particular unlikely to crash, because unmarshalling will always catch if the field is missing.

But for safety, yes -- I'll add errors proactively

Reply:
Let's avoid panic, especially if it's run as a service in the long-term, we want to always return cleanly